File: src/main/java/game/RecordStatistic.java
package game;

import game.lottery.LotteryRecord;

public class RecordStatistic extends Statistic {

	public LotteryRecord record;

	/** Number of <tt>index + 2</tt> consecutives. */
	public final int[] consecutive;

	/** Gap between ball at <tt>index</tt> and ball at <tt>index + 1</tt>. */
	public final int[] gap;

	/** Number of odd-numbered balls. */
	public int odd;
	/** Number of even-numbered balls. */
	public int even;

	public RecordStatistic(int size) {
		super();
		this.consecutive = new int[size];
		this.gap = new int[size];
	}

}

File: src/main/java/com/github/pnemonic/game/RecordStatistic.kt
package com.github.pnemonic.game

import com.github.pnemonic.game.lottery.LotteryRecord

class RecordStatistic(val record: LotteryRecord) : Statistic() {

    /**
     * Number of <tt>index + 2</tt> consecutives.
     */
    val consecutive: IntArray

    /**
     * Gap between ball at <tt>index</tt> and ball at <tt>index + 1</tt>.
     */
    val gap: IntArray

    /**
     * Number of odd-numbered balls.
     */
    var odd = 0

    /**
     * Number of even-numbered balls.
     */
    var even = 0

    init {
        val sizeConsecutives = record.balls.size - 1
        consecutive = IntArray(sizeConsecutives)
        gap = IntArray(sizeConsecutives)
    }
}